### PR TITLE
Fixing problem showing terms and conditions when user has google_maps enabled

### PR DESCRIPTION
**Pull request by xavijam**
That terms block shouldn't be visible.

screen shot 2015-06-17 at 12 08 24

REVIEWER: @javierarce 
cc @rafatower

### CODE REVIEW

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4382/))

File: lib/assets/javascripts/cartodb/table/geocoding_dialog/panes/geocoding_dialog_pane_address.js [hunk 1]
```diff
@@ -186,10 +186,12 @@
     },
 
     _onEstimationChange: function() {
+      var googleUser = this.user.featureEnabled('google_maps');
+
       // Check if it is needed to accept the terms
-      if (!this.user.get('geocoding').hard_limit && ( this.estimation.get('estimation') > 0 || ( this.estimation.get('estimation') === undefined && !this.user.get('geocoding').hard_limit ))) {
+      if (!googleUser && !this.user.get('geocoding').hard_limit && ( this.estimation.get('estimation') > 0 || ( this.estimation.get('estimation') === undefined && !this.user.get('geocoding').hard_limit ))) {
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/table/geocoding_dialog/panes/geocoding_dialog_pane_address.js]: I find a bit difficult to read this condition, but I think ```!this.user.get('geocoding').hard_limit``` is duplicated and the last one could be removed.

Maybe adding variables would make this thing easier to read: 

```js
var hasHardLimit = this.user.get('geocoding').hard_limit;
var hasEstimation = this.estimation.get('estimation') !== undefined;

if (!googleUser && !hasHardLimit && (this.estimation.get('estimation') > 0 || !hasEstimation)) 

```

**Comment by javierarce** [comment]
:+1: I think the code is OK, but I added a suggestion to simplify one condition.

File: lib/assets/javascripts/cartodb/table/geocoding_dialog/panes/geocoding_dialog_pane_address.js (hunk 1 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/geocoding_dialog/panes/geocoding_dialog_pane_address.js]: It comes from the old code, but I'm going to change it, thanks!

**Comment by xavijam** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4390/))

**Pull request merged**